crates/block-producer/src/mempool/inflight_state/mod.rs

```diff
@@ -123,6 +123,14 @@ impl InflightState {
     }
 
     fn verify_transaction(&self, tx: &AuthenticatedTransaction) -> Result<(), AddTransactionError> {
+        // Check that the transaction hasn't already expired.
+        if tx.expires_at() <= self.chain_tip {
```

Comment:
Should we be a bit more aggressive here? For example, a transaction which expires in the next block is almost certainly not going to make it as we need to put it into the batch first and then the batch needs to make it into the block. Maybe it could be something like:
```rust
if tx.expires.at() <= self.chain_tip + 1 {
    ...
}
```
Instead of hard-coding `+1`, we could also have a constant for it.